golos.emit/src/golos.emit.cpp

```diff
@@ -98,47 +96,44 @@ void emission::emit() {
     }
     // TODO: maybe limit elapsed to avoid instant high value emission
 
-    auto p = control::get_params(_token);
-    auto narrowed = int64_t((now - s.start_time)/1000000 / p.infrate_narrowing);
-    int64_t infrate = std::max(int64_t(p.infrate_start) - narrowed, int64_t(p.infrate_stop));
+    auto pools = _cfg.get().pools.pools;
+    auto infrate = _cfg.get().infrate;
+    auto narrowed = int64_t((now - s.start_time)/1000000 / infrate.narrowing);
+    int64_t inf_rate = std::max(int64_t(infrate.start) - narrowed, int64_t(infrate.stop));
 
     auto token = eosio::token(config::token_name);
     auto new_tokens =
-        token.get_supply(_token.name()).amount * infrate * time_to_blocks(elapsed)
+        token.get_supply(_token.name()).amount * inf_rate * time_to_blocks(elapsed)
```

Comment:
we are multiplying three potentially large numbers, it seems that overflow is possible here.

Reply:
you're right, fixed